Sources/DeepLearningContrib/MNISTdataset.swift

```diff
@@ -0,0 +1,86 @@
+import Foundation
+import TensorFlow
+
+/// MNIST constants
+fileprivate enum constants {
+    static let URL = "http://yann.lecun.com/exdb/mnist/"
+    static let trainImagesFile = "train-images-idx3-ubyte.gz"
+    static let trainLabelsFile = "train-labels-idx1-ubyte.gz"
+    static let testImagesFile = "t10k-images-idx3-ubyte.gz"
+    static let testLabelsFile = "t10k-labels-idx1-ubyte.gz"
+    static let trainImages = "train-images-idx3-ubyte"
+    static let trainLabels = "train-labels-idx1-ubyte"
+    static let testImages = "t10k-images-idx3-ubyte"
+    static let testLabels = "t10k-label-idx1-ubyte" 
+}
+
+/// To download and extract files from the URL
+fileprivate func downloadAndExtract() {
+    let helperFiles = [constants.trainImagesFile,
+                       constants.trainLabelsFile,
+                       constants.testImagesFile,
+                       constants.testLabelsFile]
+    let extractedFiles = [constants.trainImages,
+                          constants.trainLabels,
+                          constants.testImages,
+                          constants.testLabels]
+    var counter = 0
+    let path = FileManager.default.currentDirectoryPath
+    for helperFile in helperFiles {
+        let url = NSURL(fileURLWithPath: path)
+        if let pathComponent = url.appendingPathComponent(helperFile) {
```

Comment:
With `URL`, this is non-optional, so it should not require an `if let`!

Also I think `pathComponent` should be named something like `helperFileLocalUrl`, to describe that it's the local system url where the file should be.